packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -156,9 +160,10 @@ class AnimatedSwitcher extends StatefulWidget {
   /// The animation curve to use when transitioning the previous [child] out.
   final Curve switchOutCurve;
 
-  /// A function that wraps the new [child] with an animation that transitions
+  /// A function that wraps a new [child] with an animation that transitions
   /// the [child] in when the animation runs in the forward direction and out
-  /// when the animation runs in the reverse direction.
+  /// when the animation runs in the reverse direction. This is only called
+  /// when a new [child] is set (not for each build).
```

Comment:
it should presumably be set any time the builder itself is changed, no?

ideally this should have the same semantics as the `Builder` widget.

Reply:
> Are you saying that all the previous children should have their transitions updated when the transitionBuilder changes?

Yeah.